feature-based-molecular-networking/tools/feature-based-molecular-networking/scripts/progenesis_formatter.py

```diff
@@ -147,17 +150,36 @@ def convert_mgf(input_msp, output_mgf, compound_to_scan_mapping):
 
             read_name = True
             scan = (compound_to_scan_mapping[compound_name])
-        elif line.startswith("PrecursorMZ:"):
-            precursor_mz = (line.rstrip().replace("PrecursorMZ: ", ""))
-        elif line.startswith("Charge:"):
-            charge = (line.rstrip().replace("Charge: ", ""))
-        elif len(line.rstrip()) == 0 and read_name == True:
+
+        if line.startswith("PrecursorMZ:"):
+            precursor_mz = (line.rstrip().replace("PrecursorMZ: ", "PEPMASS="))
+        if line.startswith("Charge:"):
+            charge = []
+            charge = (line.rstrip().replace("Charge: ", "CHARGE="))
+            precursor_type = []
+        elif line.startswith("Precursor_type:"):
```

Comment:
Is this for positive and negative? If so can add comment?

Reply:
it works for both. I am adding comments.